packages/boot/src/components/app/router.tsx

```diff
@@ -7,43 +7,144 @@ import {
 	createRoute,
 	RouterProvider,
 	createBrowserHistory,
+	type AnyRoute,
 } from '@tanstack/react-router';
 import { parseHref } from '@tanstack/history';
+import type { ComponentType } from 'react';
 
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { lazy, Suspense, useState, useEffect } from '@wordpress/element';
+import { Page } from '@wordpress/admin-ui';
 
 /**
  * Internal dependencies
  */
 import Root from '../root';
-import Home from '../home';
+import type { Route, RouteLoaderContext } from '../../store/types';
 
 // Not found component
 function NotFoundComponent() {
 	return (
 		<div className="boot-layout__stage">
-			<h1>{ __( 'Route not found' ) }</h1>
+			<Page title={ __( 'Route not found' ) } hasPadding>
+				{ __( "The page you're looking for does not exist" ) }
+			</Page>
 		</div>
 	);
 }
 
-// Create root route
-const rootRoute = createRootRoute( {
-	component: Root,
-} );
+function RouteComponent( {
+	stage: Stage,
+	inspector: Inspector,
+}: {
+	stage?: ComponentType;
+	inspector?: ComponentType;
+} ) {
+	return (
+		<>
+			{ Stage && (
+				<div className="boot-layout__stage">
+					<Suspense fallback={ <div>Loading...</div> }>
```

Comment:
Shouldn't we make sure this component doesn't load at all by putting the dependencies/imports in the loader function?

Reply:
We should, right now I don't have any thing like that in my routes, so you won't see it. Do you think I should remove the fallback?